### PR TITLE
arch/arm/src/amebad: Update serial_api

### PR DESCRIPTION
**Pull request by AdamZhang0124**
This commit updates serial_init process and removes the redundant code.

### CODE REVIEW

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 1]
```diff
@@ -417,7 +418,6 @@ static int rtl8721d_up_setup(struct uart_dev_s *dev)
 	serial_baud(sdrv[uart_index_get(priv->tx)], priv->baud);
 	serial_format(sdrv[uart_index_get(priv->tx)], priv->bits, priv->parity, priv->stopbit);
 	serial_set_flow_control(sdrv[uart_index_get(priv->tx)], priv->FlowControl, priv->rts, priv->cts);
-	serial_enable(sdrv[uart_index_get(priv->tx)]);
```
Comment (jeongchanKim) [inline os/arch/arm/src/amebad/amebad_serial.c:420]: The only difference in changes is removing this line. Other changes are redundant code or tidy up the code. 
Is it okay we don't have this line?

Comment (AdamZhang0124) [reply os/arch/arm/src/amebad/amebad_serial.c:420]: Yes, the corresponding peripheral clock functions are enabled in the bootloader, therefore it is redundant to enable again at here. Thanks!

Comment (jeongchanKim) [reply os/arch/arm/src/amebad/amebad_serial.c:420]: Ok. Thanks for checking. I will approve it.

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Comment by sunghan-chang** [comment]
Because we use uart1 on internal usages and it's much important peri, let's check first.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**